### DISCUSSION TITLE
how to save image?

### DISCUSSION THREAD

**Discussion by david-shu** [flet-dev/flet#1447] [Q&A] [answered]
how to save img in flet app?

```
def save(e):
    p =  e.control.page
    p.launch_url( e.control.content.src )

img = Image( src='aa.png')
c = Container( content=img, on_click=save )
```

**Comment by FeodorFitsner** [comment]
You are working in Python, just download the image using any Python HTTP library and save it to a file.

**Reply by david-shu** [reply]
Sorry for being a bit confused. How can I use Python to click on an image in the browser client and save it to the client?

```
  def save(e):
     import requests
    imgurl = e.control.content.src
    r = requests.get( imgurl )
    with open( 'xxx.png', 'wb') as f:
          f.write( r.content )
   # but this image is only downloaded to the server side
  
  img = Image( src='aa.png')
  c = Container( content=img, on_click=save )
```

**Reply by aedhcarrick** [reply]
I would like to bump this request.  Perhaps adding a serve_content.go to the server package.  Not sure if I am phrasing that correctly.  I have no experience with Go, but think it should be possible?
Flet is awesome, but many apps perform some sort of file manipulation.  It would be nice if web users could download the results.  Seems redundant to use flask or fastapi as a backend to the backend, just to serve files.

**Reply by FeodorFitsner** [reply]
It's already possible. I'm not sure I understand the issue. You can put files that must be downloadable to `assets` directory and then present user a link or perform redirect to that file to cause download. Could you please elaborate what are you trying to achieve?

**Reply by aedhcarrick** [reply]
apologies for the late reply.
if this seems like a noob question... it is. imma a total noob when it comes to web dev.
i tried a redirect like you suggest, but get varied results, with different file types.
i made a short example here--> https://github.com/aedhcarrick/test-dl
seems the browser tries to open the file when the link is opened. in the case of zip files, it downloads, cause it doesn't know what else to do, but chrome at least, doesn't give the option of where to save.  also, I haven't tested yet, but i can foresee problems occurring with large (>7.5GB) downloads or even multiple concurrent ones. I was thinking of an approach like the one mentioned in the answer to this question on stack overflow.--> https://stackoverflow.com/a/24116517
Would be nice to modify header content and stream the response body (in case of large files).
apologies again, if i am just further confusing the issue.

**Reply by aedhcarrick** [reply, marked as answer]
if i understand correctly, marking the file as an attachment (in the header), would trigger the browser's native file dialog.
